gadopt/stokes_integrators.py

```diff
@@ -78,20 +79,26 @@ def create_stokes_nullspace(Z, closed=True, rotational=False, translations=None)
         for tdim in translations:
             vec = [0] * dim
             vec[tdim] = 1
-            basis.append(fd.Function(V).interpolate(fd.as_vector(vec)))
+            basis.append(fd.Function(stokes_funcspace[0]).interpolate(fd.as_vector(vec)))
 
     if basis:
         V_nullspace = fd.VectorSpaceBasis(basis, comm=Z.mesh().comm)
         V_nullspace.orthonormalize()
     else:
-        V_nullspace = V
+        V_nullspace = stokes_funcspace[0]
 
     if closed:
         p_nullspace = fd.VectorSpaceBasis(constant=True, comm=Z.mesh().comm)
     else:
-        p_nullspace = W
+        p_nullspace = stokes_funcspace[1]
 
-    return fd.MixedVectorSpaceBasis(Z, [V_nullspace, p_nullspace])
+    null_space = [V_nullspace, p_nullspace]
+
+    # If free surface unknowns, add dummy free surface nullspace
+    for i in range(len(stokes_funcspace)-2):
+        null_space.append(stokes_funcspace[2+i])
```

Comment:
Not actually sure if this works, because maybe `subfunctions` doesn't quite return a list?

```suggestion
    # If free surface unknowns, add dummy free surface nullspace
    null_space += stokes_funcspace[2:]
```

Hopefully this idea of extra function spaces doesn't clash with any other physics later on!

Reply:
yeah I guess this might not work if other function spaces also require nullspaces but I guess we can change that down the line.